drivers/SmartThings/zwave-switch/src/inovelli-vzw32-sn/init.lua

```diff
@@ -0,0 +1,389 @@
+        device:send(SwitchMultilevel:Get({}))
+      end)
+    else
+      device:emit_event(capabilities.switch.switch(value == 0 and "off" or "on"))
+      local dev = device:get_parent_device()
+      local config = Configuration:Set({
+        parameter_number=NOTIFICATION_PARAMETER_NUMBER,
+        configuration_value=(value == 0 and 0 or getNotificationValue(device)),
+        size=4
+      })
+      local send_configuration = function()
+        dev:send(config)
+      end
+      device.thread:call_with_delay(1,send_configuration)
+    end
+  end
+end
+
+local function version_report(driver, device, cmd)
+  log.info("Firmware Version: "..cmd.args.firmware_0_version.."."..cmd.args.firmware_0_sub_version)
+end
+
+local map_key_attribute_to_capability = {
+  [CentralScene.key_attributes.KEY_PRESSED_1_TIME] = capabilities.button.button.pushed,
+  [CentralScene.key_attributes.KEY_RELEASED] = capabilities.button.button.held,
+  [CentralScene.key_attributes.KEY_HELD_DOWN] = capabilities.button.button.down_hold,
+  [CentralScene.key_attributes.KEY_PRESSED_2_TIMES] = capabilities.button.button.pushed_2x,
+  [CentralScene.key_attributes.KEY_PRESSED_3_TIMES] = capabilities.button.button.pushed_3x,
+  [CentralScene.key_attributes.KEY_PRESSED_4_TIMES] = capabilities.button.button.pushed_4x,
+  [CentralScene.key_attributes.KEY_PRESSED_5_TIMES] = capabilities.button.button.pushed_5x,
+}
+
+local function central_scene_notification_handler(self, device, cmd)
+  if ( cmd.args.scene_number ~= nil and cmd.args.scene_number ~= 0 ) then
+    local button_number = cmd.args.scene_number
+    local capability_attribute = map_key_attribute_to_capability[cmd.args.key_attributes]
+    local additional_fields = {
+      state_change = true
+    }
+
+    local event
+    if capability_attribute ~= nil then
+      event = capability_attribute(additional_fields)
+    end
+
+    if event ~= nil then
+      -- device reports scene notifications from endpoint 0 (main) but central scene events have to be emitted for button components: 1,2,3
+      local comp = device.profile.components[button_to_component(button_number)]
+      if comp ~= nil then
+        device:emit_component_event(comp, event)
+      end
+    end
+  end
+end
+
+local function basic_and_switch_binary_report_handler(driver, device, cmd)
+  local value = cmd.args.target_value and cmd.args.target_value or cmd.args.value
+  local event = value == SwitchBinary.value.OFF_DISABLE and capabilities.switch.switch.off() or capabilities.switch.switch.on()
+  device:emit_event_for_endpoint(cmd.src_channel, event)
+end
+
+local function onoff_level_report_handler(self, device, cmd)
+  local value = cmd.args.target_value and cmd.args.target_value or cmd.args.value
+  device:emit_event(value == SwitchMultilevel.value.OFF_DISABLE and capabilities.switch.switch.off() or capabilities.switch.switch.on())
+  if value >= 0 then
+    device:emit_event(capabilities.switchLevel.level(value >= 99 and 100 or value))
+  end
+end
```

Comment:
These should already be handled by our defaults inherited from the parent driver, I expect.